### PR TITLE
[Merged by Bors] - feat: `IsScalarTower` and `SMulCommClass` instances for pointwise actions of submodules

### PR DESCRIPTION
**Pull request by YaelDillies**
Moves:
- `Submodule.smul_def` → `Submodule.setSemiring_smul_def`

From GrowthInGroups

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Algebra/Operations.lean [hunk 1]
```diff
@@ -635,7 +656,7 @@ instance moduleSet : Module (SetSemiring A) (Submodule R A) where
 
 variable {R A}
 
-theorem smul_def (s : SetSemiring A) (P : Submodule R A) :
+theorem setSemiring_smul_def (s : SetSemiring A) (P : Submodule R A) :
```
Comment (eric-wieser) [inline Mathlib/Algebra/Algebra/Operations.lean:659]: Please list this in the "moves" section of the PR template.

File: Mathlib/Algebra/Algebra/Operations.lean [hunk 2]
```diff
@@ -353,6 +355,25 @@ theorem comap_op_mul (M N : Submodule R Aᵐᵒᵖ) :
         comap (↑(opLinearEquiv R : A ≃ₗ[R] Aᵐᵒᵖ) : A →ₗ[R] Aᵐᵒᵖ) M := by
   simp_rw [comap_equiv_eq_map_symm, map_unop_mul]
 
+section
+variable {B : Type*} [Semiring B] [Algebra R B] [Module A B] [SMulCommClass A R B]
+
+instance [IsScalarTower A B B] : IsScalarTower A (Submodule R B) (Submodule R B) where
```
Comment (eric-wieser) [inline Mathlib/Algebra/Algebra/Operations.lean:361]: I'm pretty sure you don't need `Algebra R A` here? That is, you've taken more assumptions than you need by reusing `A` for the wrong thing. Can you write this `IsScalarTower S (Submodule R A) (Submodule R A)` for a new variable `S`?

Comment (eric-wieser) [reply Mathlib/Algebra/Algebra/Operations.lean:361]: `variable {S : Type*} [Monoid S] [DistribMulAction S A] [SMulCommClass S R A]` seems to suffice

Comment (eric-wieser) [reply Mathlib/Algebra/Algebra/Operations.lean:361]: (ok, maybe `S` is a bad name after all, but you could pick a greek letter or a subscripted `R`)

Comment (YaelDillies) [reply Mathlib/Algebra/Algebra/Operations.lean:361]: Oh, I meant for `A` to be a `R`-algebra rather than the other way around

**Comment by eric-wieser** [comment]
bors d+

Please merge if you're happy with the commits I pushed.

**Comment by YaelDillies** [comment]
Yep, looks good thanks!

bors merge

**Pull request closed without merging**